opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java

```diff
@@ -98,8 +96,8 @@ private HttpURLConnection initializeHttp(String requestURLPath, boolean useBasic
             OpensrpSSLHelper opensrpSSLHelper = new OpensrpSSLHelper(context, configuration);
             ((HttpsURLConnection) urlConnection).setSSLSocketFactory(opensrpSSLHelper.getSSLSocketFactory());
         }
-        urlConnection.setConnectTimeout(CONNECTION_TIMEOUT);
-        urlConnection.setReadTimeout(READ_TIMEOUT);
+        urlConnection.setConnectTimeout(CoreLibrary.getInstance().getSyncConfiguration().getConnectionTime());
```

Comment:
Because this settings affect any HTTP calls made using this class and not necessarily sync why not maintain these settings on this class

Reply:
HttpAgent should ideally be accessed from within the OpenSRP Context class.

Setting the properties on the HttpAgent itself would mean somehow passing a configuration object to the OpenSRP Context class that initializes the HttpAgent.

Instead of creating an additional object to do this configuration, why not use the existing SyncConfiguration? The properties are still related to sync and it is a less drastic change.

What was your idea on how to do this?

Reply:
Defining the timeouts on `HTTPAgent` and then defining setter methods on the HTTPAgent class. 

Since `HTTPAgent` is a singleton instance it can be changed after CoreLibrary initialization

Reply:
Okay, it's not  exactly a singleton but that works.

Reply:
Cool, kept both the option to configure timeouts using the http agent and also using the sync configuration object.